Lib/test/test_socket.py

```diff
@@ -3394,7 +3395,7 @@ def testCMSG_SPACE(self):
         # Test CMSG_SPACE() with various valid and invalid values,
         # checking the assumptions used by sendmsg().
         toobig = self.socklen_t_limit - socket.CMSG_SPACE(1) + 1
-        values = list(range(257)) + list(range(toobig - 257, toobig))
+        values = list(range(257)) + list(range(toobig - 257, toobig - 8))
```

Comment:
This is a 'fix' we are using internally, probably not acceptable (more in the issue tracker).

Reply:
What are the values of `socklen_t_limit` and `socket.CMSG_SPACE(1)` on Solaris?

Reply:
Well, I think that the simplest way to resolve this is to add Solaris-specific fix:
```suggestion
        if SOLARIS:
            toobig -= 8
        values = list(range(257)) + list(range(toobig - 257, toobig))
```

Reply:
The values of `socklen_t_limit` and `socket.CMSG_SPACE(1)` are here:
```python
>>> import _testcapi
>>> import socket
>>> _testcapi.INT_MAX
2147483647
>>> socket.CMSG_SPACE(0)
16
>>> socket.CMSG_SPACE(1)
16
```
These are Solaris SPARC specific; on intel it's pretty similar to Linux.